src/dune_rules/jsoo/jsoo_rules.ml

```diff
@@ -275,9 +337,21 @@ let js_of_ocaml_rule
        | None -> S []
        | Some config ->
          Dyn
-           (Action_builder.map config ~f:(fun config ->
-              Command.Args.S
-                (List.map (Config.to_flags config) ~f:(fun x -> Command.Args.A x)))))
+           (let+ config = config
+            and+ jsoo_version =
+              let* jsoo = jsoo in
+              Action_builder.of_memo (Version.jsoo_version jsoo)
+            in
+            let recent =
```

Comment:
The name `recent` does not say much . I think I would rather pass the version down and compare against it when needed.